File: Algorithms/src/3550. Smallest Index With Digit Sum Equal to Index/Code/solution.cpp
class Solution {
public:
    int smallestIndex(vector<int>& nums) {
        for (int i = 0; i < nums.size(); ++i) {
            int x = nums[i];
            int sum = 0;
            while (x > 0) {
                sum += x % 10;
                x /= 10;
            }
            if (sum == i) return i;
        }
        return -1;
    }
};

File: Algorithms/src/3550. Smallest Index With Digit Sum Equal to Index/Code/solution.ts
function smallestIndex(nums: number[]): number {
    for (let i = 0; i < nums.length; ++i) {
        let x = nums[i];
        let sum = 0;
        while (x > 0) {
            sum += x % 10;
            x = Math.floor(x / 10);
        }
        if (sum === i) return i;
    }
    return -1;
};